tests/Nidup/Architool/Infrastructure/Filesystem/FolderReferenceUpdaterTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+use PHPUnit\Framework\TestCase;
+use org\bovigo\vfs\vfsStreamWrapper;
+use org\bovigo\vfs\vfsStream;
+use org\bovigo\vfs\vfsStreamDirectory;
+use Nidup\Architool\Infrastructure\Filesystem\FolderReferenceUpdater;
+use Nidup\Architool\Domain\Model\Folder;
+use Nidup\Architool\Domain\Model\Folder\FolderNamespace;
+
+class FolderReferenceUpdaterTest extends TestCase
+{
+
+    private $root;
+
+    public function setUp()
+    {
+        vfsStreamWrapper::register();
+
+        $rootStructure = [
+            'My' => [
+                'Project' => [
+                    'Dir' => [
+                        'features' => [],
+                        'tests' => [],
+                        'app' => [
+                            'config' => []
+                        ],
+                        'src' => [
+                            'Akeneo' => [
+                                'Component' => [
+                                    'Batch' => [],
+                                    'BatchQueue' => [],
+                                ],
+                                'Pim' => [
+                                    'Platform' => [
+                                        'Domain' => [
+                                            'Batch' => []
+                                        ]
+                                    ]
+                                ]
+                            ]
+                        ]
+                    ]
+                ]
+            ]
+        ];
+
+        $this->root = vfsStream::setup('root', null, $rootStructure);
+    }
+
+    public function testNamespaceDeclarationIsUpdatedInPhp()
```

Comment:
youShouldNotUseCamelCaseBecauseICanEasilyUnderstandTheUseCaseYouAreDescribing

Reply:
If I correctly understand, you are testing that "a class namespace is well updated when you move it" ?

Reply:
Yes, that what the test is called...